padle/padle.py

```diff
@@ -252,20 +253,18 @@ async def stats(self, ctx):
                 correct_id = await self.config.padle_today()
             else:
                 correct_id = save_daily[day - 1][0]
-            if str(day) in all_guesses and len(all_guesses[str(day)]) != 0 and all_guesses[str(day)][-1] == correct_id:
+            if str(day) in all_guesses and correct_id in all_guesses[str(day)]:
                 cur_streak += 1
                 wins += 1
-            else:
-                if cur_streak > max_streak:
-                    max_streak = cur_streak
-                if day != await self.config.num_days():
-                    cur_streak = 0
-
-        if cur_streak > max_streak:
-            max_streak = cur_streak
+            elif day != await self.config.num_days() or await self.config.user(ctx.author).done():
+                cur_streak = 0
+            if cur_streak > max_streak:
+                max_streak = cur_streak
```

Comment:
Just say `max_streak = max(max_streak, cur_streak)`.

You can also place it in the `if (correct)` block, because you never need to update max_streak if current_streak didn't increase. (You can place it in the `elif` block, right before you reset the streak, but then you'd have to add one more `max` after the loop to catch the ending streak.)